programs/eosio-launcher/main.cpp

```diff
@@ -798,7 +820,7 @@ launcher_def::generate () {
       sf.close();
     }
     if (host_map_file.empty()) {
-      savefile = bfs::path (output.stem().string() + "_hosts.json");
+      savefile = bfs::path (bfs::path(cluster_id_str()) / (output.stem().string() + "_hosts.json"));
     }
     else {
       savefile = bfs::path (host_map_file);
```

Comment:
Here you could bring in `cluster_id_str()` since `host_map_file` wouldn't be modified earlier. But more importantly, why was this being done in the original code in the first place. It seems like a bad idea for the input to also maybe be an output file.

Reply:
changed it to ```savefile = cluster_dir_path() / (output.stem().string() + "_hosts.json"));```